### PR TITLE
[JENKINS-48917] Add option to ignore LDAP servers if they are unavailable.

### PR DESCRIPTION
**Pull request by dwnusbaum**
See [JENKINS-48917](https://issues.jenkins-ci.org/browse/JENKINS-48917).

This change allows communication failures with LDAP servers (i.e., DataAccessException, AuthenticationServiceException) to be ignored so that the failed action will be reattempted with the next configuration.

@reviewbybees

### CODE REVIEW

File: src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/config.jelly [hunk 1]
```diff
@@ -47,5 +47,8 @@
                 </f:entry>
             </f:repeatableProperty>
         </f:entry>
+        <f:entry field="ignoreIfUnavailable" title="${%Ignore if Unavailable}">
+            <f:checkbox />
+        </f:entry>
     </f:advanced>
 </j:jelly>
```
Comment (Wadeck) [inline src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/config.jelly]: just a line break at the end could be nice

File: src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    <p>
+    If checked, this option indicates that authentication and user/group membership lookups
+    will be attempted against subsequent LDAP domains even if this domain is unavailable.
+    <p>
+    If unchecked (the default), authentication and user/group memberships for users in
+    subsequent LDAP domains will fail if this domain is unavailable.
+    <p>
+    The latter is useful when there is a concern that the same username may exist
+    across domains and should not map to the same Jenkins user.
+</div>
```
Comment (Wadeck) [inline src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html]: ditto

File: src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<div>
+    <p>
```
Comment (Wadeck) [inline src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html]: quite weird to not close the `p` tag but seems to case in each help file in the project. I would use `<br />` instead personnally to be valid with the standard.

File: src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    If checked, this option indicates that authentication and user/group membership lookups
+    will be attempted against subsequent LDAP domains even if this domain is unavailable.
```
Comment (Wadeck) [inline src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html]: _perhaps just my english level_ 
this sentence seems to say that if the domain is available and respond correctly the subsequent domains will *still* be lookup.
From my understanding of the initial problem, the goal is to let administrators say that a domain is somewhat optional, if it fails, we try the next one. If it succeeds, we use that information.

File: src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    <p>
+    If unchecked (the default), authentication and user/group memberships for users in
+    subsequent LDAP domains will fail if this domain is unavailable.
```
Comment (Wadeck) [inline src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html]: > will fail

"will not even be attemped" no ?

Comment (dwnusbaum) [reply src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html]: Yes I will reword this, I should also change `memberships` to `membership lookups`. Thanks!

File: src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html (hunk 4 above)
Comment (dwnusbaum) [reply src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html]: Good point, I will try reword this to make it very clear that it only happens in the event of failure.

File: src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html (hunk 3 above)
Comment (dwnusbaum) [reply src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/help-ignoreIfUnavailable.html]: Yeah I had `</p>` tags at first but removed them to be consistent across the plugin. I went ahead and just added them back in 5c1c278.

File: src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/config.jelly (hunk 1 above)
Comment (dwnusbaum) [reply src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/config.jelly]: Since this isn't a new file, my preference is to just leave it as-is and handle it in a formatting-only PR later if desired. Is that ok?

Comment (Wadeck) [reply src/main/resources/jenkins/security/plugins/ldap/LDAPConfiguration/config.jelly]: My concern was more concerning the help text, so no problem

**Comment by dwnusbaum** [comment]
@reviewbybees done
